spec/models/order_spec.rb

```diff
@@ -724,11 +724,128 @@
     end
   end
 
-  describe "Callbacks" do
+  describe ".assign_code" do
     let(:order) { create(:order, :with_state_draft, :with_orders_packages) }
 
-    it "Assigns GC Code" do
-      expect(order.code).to include("GC-")
+    context 'Goodcity orders' do
+      it "Assigns GC Code" do
+        expect(order.code).to include('GC-')
+      end
+    end
+
```

Comment:
This is good testing but the spec file is super long. Can you extract these out into `spec/models/concerns/order_code_generator_spec.rb`?